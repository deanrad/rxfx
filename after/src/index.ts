export { concat, EMPTY, throwError } from 'rxjs';
export * from './after';
