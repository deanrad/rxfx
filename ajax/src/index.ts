export { ajax } from 'rxjs/ajax';
export * from './fetchMany';
export * from './makeAbortableHandler';
export * from './makeSSEHandler';