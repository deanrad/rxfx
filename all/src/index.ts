export * from './after';
export * from './bus';
export * from './fsa';
export * from './perception';
export * from './service';
export * from './effect';

export { Subscription, ObservableInput, interval, from } from 'rxjs';
export * from 'rxjs/ajax';
