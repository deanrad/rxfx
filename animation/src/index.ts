export * from '@rxfx/after';
export * from '@rxfx/perception';
export * from './tweens';
export * from './tweenjs';
export * from './custom';
