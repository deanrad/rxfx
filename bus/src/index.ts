export * from './bus';
export * from './trace';
