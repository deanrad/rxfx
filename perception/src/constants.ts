// See https://dev.to/deanius/the-thresholds-of-perception-in-ux-435g
export const THRESHOLD = {
  Frame: 16,
  Frame60: 16,
  Frame90: 11,
  Unison: 10,
  Chorus: 25,
  MovieFrame: 42,
  Echo: 100,
  Blink: 150,
  TypingKey90: 150,
  AnimationShort: 200,
  TypingKey48: 250,
  Debounce: 330,
  TypingKeyMobile: 330,
  AnimationLong: 400,
  EDMBeat: 500,
  DoubleClick: 500,
  Thought: 1000,
  PageLoadMax: 2000,
  DeepBreath: 4000,
  Sentence: 5000,
};

/** The 'ideal' debounce time- shorter than a long animation. About the
 * time it takes to type a key on a mobile interface.
 */
export const DEBOUNCE_DELAY = THRESHOLD.TypingKeyMobile;

/** The 'ideal' throttle time - long enough to catch most double-clicks */
export const THROTTLE_DELAY = THRESHOLD.DoubleClick;
