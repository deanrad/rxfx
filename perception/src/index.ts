export * from './constants';
export * from './randomizer';
