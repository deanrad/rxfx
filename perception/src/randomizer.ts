/** Randomizes the timing of events, preserving the average duration.
 * @see Poisson Process
 * @link https://codesandbox.io/s/rxfx-example-poisson-process-typing-hncrl0 */
export function randomizePreservingAverage(duration: number) {
  return duration * Math.log(Math.random()) * -1;
}
