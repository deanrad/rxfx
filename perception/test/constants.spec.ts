import { DEBOUNCE_DELAY, THRESHOLD, THROTTLE_DELAY } from '../src/constants';

describe('UX Constants', () => {
  it('defines thresholds of perception', () => {
    expect(THRESHOLD).toMatchSnapshot();
  });
  it('defines an ideal debounce delay', () => {
    expect(DEBOUNCE_DELAY).toMatchSnapshot();
  });
  it('defines an ideal throttle delay', () => {
    expect(THROTTLE_DELAY).toMatchSnapshot();
  });
});
