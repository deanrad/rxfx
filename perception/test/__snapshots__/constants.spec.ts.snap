// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UX Constants defines an ideal debounce delay 1`] = `330`;

exports[`UX Constants defines an ideal throttle delay 1`] = `500`;

exports[`UX Constants defines thresholds of perception 1`] = `
{
  "AnimationLong": 400,
  "AnimationShort": 200,
  "Blink": 150,
  "Chorus": 25,
  "Debounce": 330,
  "DeepBreath": 4000,
  "DoubleClick": 500,
  "EDMBeat": 500,
  "Echo": 100,
  "Frame": 16,
  "Frame60": 16,
  "Frame90": 11,
  "MovieFrame": 42,
  "PageLoadMax": 2000,
  "Sentence": 5000,
  "Swipe": 1000,
  "Thought": 1000,
  "TypingKey48": 250,
  "TypingKey90": 150,
  "TypingKeyMobile": 330,
  "Unison": 10,
}
`;
